spec/integration/aws_security_group_spec.rb

```diff
@@ -121,35 +127,343 @@
         ).and be_idempotent
       end
 
-      it "aws_security_group 'test_sg' with inbound and outbound rules works" do
+      it "can specify rules as a mapping from source/destination to port and protocol" do
         expect_recipe {
           aws_security_group 'test_sg' do
+            # We need to define a list of ports and its easier to use a method than
+            # have to add a new number when changing this test
+            def p()
```

Comment:
Great maintenance enhancer.  Two clarity suggestions:

1. These look like final IP quads, rather than port numbers?
1. A name more descriptive than `p`.